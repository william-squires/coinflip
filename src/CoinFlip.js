import { useState } from "react";
import Coin from "./Coin"


/** Flips a coin and logs how many of each side.
 * 
 * props
 * - none
 * 
 * state
 * - numHeads
 * - numTails
 * - face
 */
function CoinFlip() {
    const [numHeads, setNumHeads] = useState(0)
    const [numTails, setNumTails] = useState(0)
    const [face, setFace] = useState()
    let numFlips = numHeads + numTails


    /** returns heads or tails randomly */
    function flip() {
        const rand = Math.floor(Math.random() * 2)
        return rand === 1 ? "heads" : "tails"
    }

    function handleClick(evt) {
        setFace(flip());
        if (face === "heads") {
            setNumHeads(numHeads + 1)
        } else {
            setNumTails(numTails + 1)
        }

    }

    return ( 
    <div>
        <h1>Flip A Coin</h1>
        <div style={{visibility: "hidden"}}>
            <Coin face={face}/>
        </div>
        
        <button onClick={handleClick}>Flip!</button>
        <p>Out of {numFlips}, there have been {numHeads} heads and {numTails} tails.</p>
    </div>)
}

export default CoinFlip;