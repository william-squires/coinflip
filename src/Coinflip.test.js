import React from "react";
import { render } from "@testing-library/react";
import CoinFlip from "./CoinFlip";

it("smoketest - renders without crashing", function () {
    render(<CoinFlip />);
});
// end

it("matches snapshot", function () {
    const { container } = render(
        <CoinFlip />);
    expect(container).toMatchSnapshot();
});