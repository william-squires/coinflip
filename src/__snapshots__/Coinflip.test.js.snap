// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div>
    <h1>
      Flip A Coin
    </h1>
    <div
      style="visibility: hidden;"
    >
      <img
        src="undefined.png"
      />
    </div>
    <button>
      Flip!
    </button>
    <p>
      Out of 
      0
      , there have been 
      0
       heads and 
      0
       tails.
    </p>
  </div>
</div>
`;
